hazelcast-jet-core/src/main/java/com/hazelcast/jet/Job.java

```diff
@@ -252,7 +257,19 @@ default void join() {
      * won't automatically delete the IMap it is exported into. You must
      * manually call {@linkplain JobStateSnapshot#destroy() snapshot.destroy()}
      * to delete it. If your state is large, make sure you have enough memory
-     * to store it.
+     * to store it. The snapshot created using this method will also not be
+     * used for automatic restart - should the job fail, the previous
+     * automatically started snapshot will be used.
+     * <p>
+     * For transactional sources or sinks (that is those which use transactions
+     * to confirm reads or to commit writes), Jet will not commit the
+     * transactions when creating a snapshot using this method. The reason for
```

Comment:
It indeed does. That's why I proposed to remove this method entirely, but we need to discuss it. 

Copying the last snapshot (if there is one) would not solve it. Correctness of 2pc requires that you restore from the latest snapshot, not some older one. Two things can happen:
- before you manage to restore from the exported snapshot, another snapshot could be done and the exported one is stale
- job fails after snapshot was exported, but before it does another automatic snapshot. It will restart from the automatic snapshot done before the exported snapshot, which is again wrong.

This method just doesn't fit and is not useful anyway, even without transactional processors.

Reply:
It's still useful for log based sources i.e. Kafka, but yes it's a fair point. It seems better to always use exportAndCancel at the same time.